CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+##################################################################
+# CMake file for Modelica Buildings Library
+# To build all targets, run
+# rm -rf build && mkdir build && cd build && cmake ../ && cmake --build . --target install && cd ..
+#
+##################################################################
+cmake_policy(SET CMP0048 NEW)
+
+project( Modelica-Buildings VERSION 5.1.0 )
+cmake_minimum_required(VERSION 3.8)
+
+# find_package(Java REQUIRED)
+# include(UseJava)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -ldl")
+
+set( BUILDINGS_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/Buildings")
+
+set( CMAKE_INSTALL_PREFIX "${BUILDINGS_INSTALL_PREFIX}/Resources/Library/${PLATFORM_INSTALL_PREFIX}/" )
+
+# Output build system directories.
+message("CMake source directory is: ${CMAKE_SOURCE_DIR}")
+
+# Detect OS.
+if(CMAKE_SYSTEM_NAME MATCHES Windows)
+    set(WINDOWS 1)
+    message("Windows system detected.")
+elseif(CMAKE_SYSTEM_NAME MATCHES Linux)
+    set(LINUX 1)
+    message("Linux system detected.")
+elseif(CMAKE_SYSTEM_NAME MATCHES Darwin)
+    set(MACOSX 1)
+    message("Mac OS X system detected.")
+else()
+    message(WARNING "OS unknown!")
+endif()
+
+# Set target directories.
+# NOTE: always add quotes to protect spaces in path when setting new variables.
+if (WINDOWS)
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES ".*64$")
```

Comment:
I think you might want to use CMAKE_SIZEOF_VOID_P https://cmake.org/cmake/help/v3.12/variable/CMAKE_SIZEOF_VOID_P.html#variable:CMAKE_SIZEOF_VOID_P because this will more accurately tell you the architecture of the target you are making as opposed to just the type of machine you are compiling on. Also I believe CMAKE_SIZEOF_VOID_P is more conventional. Relevant discussion is here http://cmake.3232098.n2.nabble.com/Detecting-32-64-bit-systems-td6019142.html